assets/resource.py

```diff
@@ -46,6 +47,26 @@ def check(self, source, version):
 
         return new_versions
 
+    def check(self, source, version):
+        """Check for new version(s)."""
+
+        index = source['index']
+        ssl_verify = source.get('ssl_verify', True)
+
+        if isinstance(ssl_verify, bool):
+            verify = ssl_verify
+        elif isinstance(ssl_verify, str):
+            verify = str(tempfile.NamedTemporaryFile(delete=False, prefix='ssl-').write(verify))
+
+        # request index
+        response = requests.request('GET', index, verify=verify)
+        response.raise_for_status()
+
+        if source.get('etag'):
+            return self.check_etag(source, version, response)
+        else:
```

Comment:
> what it means to fall back to regex version searching

Yeah, maybe something like "this checks the body of the response for something that looks like a version number"? That being said, I can't think of a real-world scenario where that would be useful, so imagine we will end up taking it out.

> I don't see a use case for fallback

Well, sites/APIs without caching support wouldn't have ETags, so would be nice to offer _something_. Did you see my fallback idea in https://github.com/concourse/archive-resource/issues/4? Ditto about the "don't worry about it for this pull request".